CONTRIBUTING.md

```diff
@@ -627,6 +631,74 @@ required.
 Essentially, the steps you have to take to deploy manually are the same that
 `skaffold` will perform, and they're outlined at the previous section.
 
+#### Kubernetes E2E tests
+
+Kubernetes integration is quite complicated, a lot of things can go wrong.
+
+To cope with the complexity and ensure we maintain high quality, we use
+E2E (end-to-end) tests.
+
+> E2E tests normally run at CI, so there's typically no need to run them
+> manually.
+
+##### Requirements
+
+* `kubernetes` cluster (`minikube` has special support, but any cluster should
+  work)
+* `docker`
+* `kubectl`
+* `bash`
+
+Vector release artifacts are prepared for E2E tests, so the ability to do that
+is required too, see Vector [docs](https://vector.dev) for more details.
+
+##### Running the E2E tests
+
+To run the E2E tests, use the following command:
+
+```shell
+CONTAINER_IMAGE_REPO=<your name>/vector-test make test-e2e-kubernetes
```

Comment:
Why not use `timberio/vector:dev` or something for a default so users don't need this?

Reply:
Users don't have push access to `timberio/vector:dev`, they'll have to specify something either way.
I was thinking about providing a default value if minikube cache is used, sane value in that case would be `localhost/<something>` due to some CRI internals.

Reply:
That sounds reasonable. :)

Reply:
Added the default when `USE_MINIKUBE_CACHE=true` is set.

Reply:
Maybe you missed it in the commit? 
![image](https://user-images.githubusercontent.com/130903/88424701-df1f7c00-cda2-11ea-8d51-0d147187329c.png)

Reply:
You still have to pass `USE_MINIKUBE_CACHE=true`.
We can actually autodetect whether `minikube` is used or now based on the `kubectl` context.
However, this would complicate the test setup a bit too much, and I chose not to do it. Besides technical reasons, the user experience won't change much, as k8s e2e tests still require a lot of involvement from the user. There was an attempt in eliminating this involvement before, with our previous k8s impl, but it wasn't very successful due to the amounts of edge cases. So, this time we took a different route and made things more robust, but less automatic.

`CONTRIBUTING.md` clearly documents the proper way of invoking the command, so this shouldn't be a problem for our users.